inc/commondbtm.class.php

```diff
@@ -862,6 +862,28 @@ function cleanDBonPurge() {
    }
 
 
+   /**
+    * Delete children items and relation with other items from database.
+    *
+    * @param array $relations_classes List of classname on which deletion will be done
+    *                                 Classes needs to extends CommonDBConnexity.
+    *
+    * @return void
+    **/
+   protected function deleteChildrenAndRelationsFromDb(array $relations_classes) {
+
+      foreach ($relations_classes as $classname) {
+         if (!is_a($classname, CommonDBConnexity::class, true)) {
+            continue;
```

Comment:
This would lend in silently failing cases... Log at least a warning.

Reply:
Done.